### PR TITLE
locationd: initial value for gps_std_factor

### PR DESCRIPTION
**Pull request by fredyshox**
This was uninitialized when using the default constructor causing undefined behaviour.

### CODE REVIEW

**Comment by haraschax** [comment]
`this->ublox_available` is a bit confusing, can you make an enum called something like gnss_source>

And then you can do
```
if (Params().getBool("UbloxAvailable", true)){
  gnss_source = UBLOX
} else {
  gnss_source = QCOM 
}
```
And then replace setup_gps with:
```
this->configure_gnss_source(gnss_source);
```

**Comment by fredyshox** [comment]
Added `LocalizerGnssSource` and `configure_gnss_source`

File: selfdrive/locationd/locationd.cc [hunk 1]
```diff
@@ -84,10 +84,10 @@ Localizer::Localizer() {
 
   VectorXd ecef_pos = this->kf->get_x().segment<STATE_ECEF_POS_LEN>(STATE_ECEF_POS_START);
   this->converter = std::make_unique<LocalCoord>((ECEF) { .x = ecef_pos[0], .y = ecef_pos[1], .z = ecef_pos[2] });
+  this->gnss_source = gnss_source;
```
Comment (haraschax) [inline selfdrive/locationd/locationd.cc:87]: Move this to configure_gnss_source?

Comment (fredyshox) [reply selfdrive/locationd/locationd.cc:87]: you're right

File: selfdrive/locationd/locationd.cc [hunk 2]
```diff
@@ -326,7 +326,7 @@ void Localizer::handle_gps(double current_time, const cereal::GpsLocationData::R
   VectorXd ecef_pos = this->converter->ned2ecef({ 0.0, 0.0, 0.0 }).to_vector();
   VectorXd ecef_vel = this->converter->ned2ecef({ log.getVNED()[0], log.getVNED()[1], log.getVNED()[2] }).to_vector() - ecef_pos;
   float ecef_pos_std;
-  if (ublox_available) {
+  if (gnss_source == LocalizerGnssSource::UBLOX) {
```
Comment (haraschax) [inline selfdrive/locationd/locationd.cc:328]: Actually while you're at it, fix this with configure_gnss_source too? You can define a variance_factor and vertical_variance factor, and set them to 0,1,3 appropriately.

File: selfdrive/locationd/locationd.cc [hunk 3]
```diff
@@ -372,7 +372,7 @@ void Localizer::handle_gnss(double current_time, const cereal::GnssMeasurements:
   }
 
   double sensor_time = log.getMeasTime() * 1e-9;
-  if (ublox_available)
+  if (gnss_source == LocalizerGnssSource::UBLOX)
```
Comment (haraschax) [inline selfdrive/locationd/locationd.cc:374]: Same here, just define a gnss_time_offset in  configure_gnss_source

**Comment by haraschax** [comment]
LGTM if tests pass

**Pull request merged**